***
# <center>Entrada y salida de datos</center>
***

## Ejercicio 001 

### Escribir un programa que permita que el usuario ingrese su nombre. El programa debe emitir una salida con un mensaje de bienvenida que incluya el nombre ingresado.
Ejemplo de ejecución:

    Ingrese su nombre: Juan
    Bienvenido Juan

In [ ]:
# Escriba la solución aquí


## Ejercicio 002 

### Escribir un programa que solicite al usuario su nombre y su edad, y luego muestre por pantalla un mensaje que diga "Hola, [nombre]. Tu edad es [edad] años."
Ejemplo de ejecución:

    Ingrese su nombre: Juan
    Ingrese su edad: 30
    Hola, Juan. Tu edad es 30 años.

In [ ]:
# Escriba la solución aquí


## Ejercicio 003

### Escribir un programa que solicite al usuario su nombre y su edad, después pida una cantidad de años y muestre por pantalla un mensaje que indique cuántos años tendrá la persona después de sumarle a su edad la cantidad de años ingresada. El mensaje debe tener el siguiente formato: 'Hola, [nombre]. Dentro de [cantidad] años tendrás [edad + cantidad] años'".
Ejemplo: Si el usuario ingresa "Juan" y "20" y luego ingresa "5", el programa debe mostrar por pantalla "Hola, Juan. Dentro de 5 años tendrás 25 años".

In [ ]:
# Escriba la solución aquí


## Ejercicio 004

### Escribir un programa que solicite al usuario ingresar tres numeros enteros.El programa debe mostrar por pantalla el resultado de sumar los tres numeros de la siguiente manera: "[numero1] + [numero2] + [numero3] = [suma]".
Ejemplo: Si el usuario ingresa 1, 2 y 3, el programa debe mostrar por pantalla: "1 + 2 + 3 = 6".

In [ ]:
# Escriba la solución aquí


## Ejercicio 005

### Escribir un programa que solicite al usuario ingresar dos notas de un alumno. El programa debe mostrar por pantalla el promedio de las notas de la siguiente manera: "Notas: [nota1] , [nota2] ==> promedio: [(nota1+nota2)/2]".
Ejemplo: Si el usuario ingresa 7 y 8, el programa debe mostrar por pantalla: "Notas: 7 , 8 ==> promedio: 7.5".

In [ ]:
# Escriba la solución aquí


## Ejercicio 006

### Escribir un programa que solicite al usuario ingresar tres notas de un alumno. El programa debe mostrar por pantalla las notas separadas por comas en un renglón y el promedio de las notas en el siguiente renglon.
Ejemplo de ejecución:
    
        Ingrese la nota 1: 7
        Ingrese la nota 2: 8
        Ingrese la nota 3: 9
        Notas: 7, 8, 9
        Promedio: 8.0

In [ ]:
# Escriba la solución aquí


## Ejercicio 007

### Escribir un programa que permita ingresar un número entero. Debe mostrarse el número ingresado:
### a. Multiplicado por 10. (utilizar el operador *)
### b. Dividido por 10. (utilizar el operador /)
### c. Elevado al cuadrado. (utilizar el operador **)
### Cada resultado debe mostrarse en una línea distinta.
Ejemplo de ejecución:

    Ingrese un número entero: 5
    5 * 10 = 50
    5 / 10 = 0.5
    5 ** 2 = 25

In [ ]:
# Escriba la solución aquí


## Ejercicio 008

### Escribir un programa que permita ingresar el valor monetario de una hora de trabajo y la cantidad de horas trabajadas por día, para calcular el salario semanal de un trabajador que trabaja todos los días hábiles y la mitad de las horas del día hábil los sábados, suponiendo que todas las horas tienen el mismo valor."

### Como pensarlo:

1- Pedir al usuario que ingrese el valor monetario de una hora de trabajo y almacenarlo en una variable valor_hora.

2- Pedir al usuario que ingrese la cantidad de horas trabajadas por día por el trabajador y almacenarla en una variable horas_trabajadas_por_dia.

3- Calcular el salario diario del trabajador multiplicando valor_hora por horas_trabajadas_por_dia.

4- Calcular el salario semanal del trabajador multiplicando el salario diario por la cantidad de días hábiles de la semana. Para esto, puedes utilizar la constante dias_habiles definida como 5.

5- Calcular la cantidad de horas trabajadas por el trabajador el sábado, que es la mitad de la cantidad de horas trabajadas por día hábil. Para esto, se puede utilizar la vaiable horas_sabado definida como horas_trabajadas_por_dia / 2.

6- Calcular el salario del trabajador por las horas trabajadas el sábado multiplicando valor_hora por horas_sabado.

7- Sumar el salario semanal con el salario del sábado para obtener el salario total semanal del trabajador.

8- Mostrar el resultado del salario semanal en la pantalla.

In [ ]:
# Escriba la solución aquí respetando los pasos descriptos en el enunciado


## Ejercicio 009

### Escribir un programa que permita ingresar valores del mismo tipo para las variables num1 y num2. Una vez cargadas, mostrar ambas variables por pantalla, intercambiá sus valores (que lo cargado en num1 quede en num2, y viceversa) y volvé a mostrarlas actualizadas.

### Como pensarlo:

1- Pedir al usuario que ingrese un valor para la variable num1.

2- Pedir al usuario que ingrese un valor para la variable num2.

3- Mostrar por pantalla el valor de las variables num1 y num2.

4- Intercambiar los valores de las variables num1 y num2.

5- Mostrar por pantalla el valor de las variables num1 y num2.   



### Otra forma de resolverlo:  
a=10  
b=20  
print(a,b)  
a = a + b;  
b = a - b;  
a = a - b;  
print(a,b)  

In [ ]:
# Escriba la solución aquí respetando los pasos descriptos en el enunciado



## Ejercicio 010

### Escribir un programa para resolver el siguiente problema:
### Tres personas invierten dinero para fundar una empresa (no necesariamente en partes iguales).

### Calcular qué porcentaje invirtió cada una.

### Como pensarlo:

- 1 Solicitar al usuario que ingrese las cantidades invertidas por cada persona en tres variables numéricas. 

   inversion_persona1 = float(input("Ingrese la cantidad invertida por la persona 1: "))
   
   inversion_persona2 = float(input("Ingrese la cantidad invertida por la persona 2: "))
   
   inversion_persona3 = float(input("Ingrese la cantidad invertida por la persona 3: "))

- 2 Calcular el total de la inversión sumando las cantidades de las tres personas.  

    total = inversion_persona1 + inversion_persona2 + inversion_persona3

- 3 Calcular el porcentaje que representa la inversión de cada persona en relación al total de la inversión.

    - 3 - 1 Dividir la cantidad invertida por cada persona entre el total de la inversión y multiplicar por 100 para obtener el porcentaje.
    Almacenar el resultado en una variable correspondiente a cada persona.
    Opcionalmente, se puede redondear el resultado a un número determinado de decimales utilizando la función round().

    Ejemplo: 

        porcentaje_inversion_persona1 = round((inversion_persona1 / total) * 100, 2)  
        porcentaje_inversion_persona2 = round((inversion_persona2 / total) * 100, 2)  
        porcentaje_inversion_persona3 = round((inversion_persona3 / total) * 100, 2)    


- 4 Mostrar por pantalla el porcentaje de inversión de cada persona.

In [ ]:
# Escriba la solución aquí respetando los pasos descriptos en el enunciado



## Ejercicio 011

### Escribir un programa que permita resolver el siguiente problema: 
### Tres personas aportan diferente capital a una sociedad y desean saber el valor total aportado y qué porcentaje aportó cada una (indicando nombre y porcentaje). 
### Solicitar la carga por teclado del nombre de cada socio y su capital aportado, a partir de esto calcular e informar lo requerido previamente.


In [ ]:
# Escriba la solución aquí


## Ejercicio 012

### Escribir un programa en Python que solicite al usuario ingresar dos valores que representen las medidas en grados de dos ángulos interiores de un triángulo. Luego, calcular y mostrar por pantalla el valor en grados del ángulo restante.
### Es importante recordar que la suma de los ángulos interiores de todo triángulo es de 180 grados. Es decir, la suma de los ángulos internos de un triángulo siempre es igual a 180 grados. Por lo tanto, para calcular el ángulo restante es necesario restar la suma de los dos ángulos interiores ingresados al valor 180."

#### Para pensar:

¿Qué pasaría si se ingresan valores negativos como medidas de ángulos?

¿Qué sucedería si la suma de los dos ángulos ingresados es mayor o igual a 180 grados?


In [ ]:
# Escriba la solución aquí


## Ejercicio 013

###  Escribir un programa para calcular el importe a cobrar por un vendedor, considerando su sueldo fijo de $200000 pesos más el 16% del monto total de ventas realizadas durante un mes.

#### Pensando los pasos para resolver el problema:

- 1 Solicitar al usuario que ingrese el monto total de ventas realizadas por el vendedor durante el 
mes en una variable correspondiente.

- 2 Calcular el 16% del monto total de ventas realizadas y almacenar el resultado en una variable.

- 3 Sumar el resultado del cálculo anterior al sueldo fijo del vendedor.

- 4 Mostrar el importe a cobrar por el vendedor.

#### Para pensar:

¿Qué pasaría si se modificara el sueldo fijo del vendedor?

Si se modifica el sueldo fijo del vendedor, entonces la fórmula utilizada para calcular el salario total debería ser actualizada para reflejar el nuevo sueldo fijo. En este caso, si el sueldo fijo aumenta, entonces el salario total también aumentaría. De igual manera, si el sueldo fijo disminuye, entonces el salario total también disminuiría. Es importante actualizar la fórmula en el programa para asegurarse de que el cálculo del salario total sea preciso y refleje el cambio en el importe a cobrar por del vendedor.

¿Hay que modificar el programa cada vez? ¿Cómo lo soluciono?

In [ ]:
# Escriba la solución aquí: (la original o la modificada)


## Ejercicio 014

### Escribir un programa que permita al usuario ingresar el ancho y largo de un terreno en metros, junto con el valor del metro cuadrado de tierra. El programa debe calcular y mostrar el valor total del terreno. Además, debe calcular la cantidad de metros de alambre necesarios para cercar completamente el terreno a tres alturas distintas.

#### Pensando los pasos para resolver el problema:

Solicitar al usuario que ingrese el ancho del terreno en metros y almacenarlo en una variable.
Solicitar al usuario que ingrese el largo del terreno en metros y almacenarlo en otra variable.
Solicitar al usuario que ingrese el valor del metro cuadrado de tierra y almacenarlo en otra variable.
Calcular el valor total del terreno multiplicando el ancho por el largo y luego multiplicando el resultado por el valor del metro cuadrado de tierra.
Mostrar el valor total del terreno al usuario.
Calcular la cantidad de metros de alambre necesarios para cercar el terreno a tres alturas distintas. Por ejemplo, se puede calcular la cantidad de alambre necesaria para cercar a 1 metro de altura, a 2 metros de altura y a 3 metros de altura. Para hacerlo, se debe sumar el perímetro del terreno (2 veces el ancho más 2 veces el largo) y luego multiplicarlo por la cantidad de alturas.
Mostrar la cantidad de metros de alambre necesarios para cercar el terreno a las tres alturas distintas al usuario.

In [ ]:
# Escriba la solución aquí

## Ejercicio 015

### Definición del problema: Una inmobiliaria paga a sus vendedores un salario base, más una comisión fija por cada venta realizada, más el 5% del valor de esas ventas. Realizar un programa que imprima el nombre del vendedor y el salario que le corresponde en un determinado mes. 
### Se leen por teclado el nombre del vendedor, la cantidad de ventas que realizó y el valor total de las mismas.

### ¿Sobran datos? ¿Qué datos sobran?

In [ ]:
# Escriba alguna solución aquí


## Ejercicio 016

### Escribir un programa que permita al usuario ingresar un período de tiempo en segundos. Luego, el programa debe convertir ese período de tiempo a una forma más legible y comprensible para el usuario, expresando el resultado en días, horas, minutos y segundos. El resultado se mostrará en pantalla en un mensaje que indique la cantidad de segundos ingresados y su equivalente en días, horas, minutos y segundos.  

Ejemplo: 200000 segundos equivalen a 2 días, 7 horas, 33 minutos y 20 segundos.

### Usar en el programa las siguientes instrucciones:

dias = segundos // 86400 # 86400 segundos = 1 día

horas = (segundos % 86400) // 3600 # 3600 segundos = 1 hora

minutos = (segundos % 3600) // 60 # 60 segundos = 1 minuto

segundos_restantes = segundos % 60 # segundos restantes


In [ ]:
# Escriba la solución aquí


## Ejercicio 017

### ¡Ayuda! Se me rompió el programa que convierte una cantidad de dinero en la cantidad mínima de billetes y monedas necesarios. Tengo todas las instrucciones necesarias, pero están todas mezcladas. ¿Podrías ayudarme a ordenarlas de manera correcta para que funcione el programa como debería? A lo mejor se me perdieron algunas instrucciones, ¿podrías agregarlas?

resto = cantidad_total  
billetes_cien = resto // 100  
resto = resto % 100  
billetes_cinco = resto // 5  
billetes_mil = resto // 1000  
billetes_cincuenta = resto // 50  
billetes_doscientos = resto // 200  
billetes_diez = resto // 10  
billetes_docientos = resto // 200  
resto = resto % 10  
cantidad_total = int(input("Ingrese la cantidad de dinero a convertir: "))  
billetes_uno = resto // 1  
print("Para la cantidad de $", cantidad_total, " se necesitan:")  
print(billetes_mil, "billetes de $1000")  
print(billetes_doscientos, "billetes de $200")  
print(billetes_cien, "billetes de $100")  
print(billetes_cincuenta, "billetes de $50")  
print(billetes_diez, "billetes de $10")  
print(billetes_cinco, "billetes de $5")  
print(billetes_uno, "billetes de $1")  


In [ ]:
# Escriba la solución aquí (¡Mucha suerte ordenando las instrucciones!)



***
# <center>Flujos de selección</center>
***

## Ejercicio 018

### Escribir un programa que permita al usuario ingresar un número entero y luego muestre un mensaje indicando si el número es par o impar.

#### Pensando los pasos para resolver el problema:

-1 Pedir al usuario que ingrese un número entero.  
-2 Almacenar ese número en una variable.  

-3 __Verificar si el número es par o impar__.  

- Si el número es par, mostrar un mensaje indicando que es par.  
- Si el número es impar, mostrar un mensaje indicando que es impar.  

(Para verificar si un número es par o impar, se puede utilizar el operador módulo (%). Si el resto de la división del número por 2 es 0, entonces el número es par. Si el resto de la división del número por 2 es 1, entonces el número es impar.)


In [1]:
# Escriba la solución aquí

numero = int(input("Ingrese un número: "))
print("El número ingresado es:", numero)
if numero % 2 == 0:
    print("El número es par")
else:
    print("El número es impar")

El número ingresado es: 123456
El número es par


## Ejercicio 019

### Escribir un programa que permita ingresar dos números enteros e indicar si son iguales o distintos.

In [ ]:
# Escriba la solución aquí


## Ejercicio 020

### Escribir un programa que permita ingresar dos cadenas de caracteres e indicar si son iguales o distintas.

In [ ]:
# Escriba la solución aquí

## Ejercicio 021

### Escribir un programa que permita ingresar dos números enteros e indicar si el primero es mayor, menor o igual al segundo.

In [ ]:
# Escriba la solución aquí


## Ejercicio 022

### Escribir un programa que permita ingresar tres números enteros e indicar cual es el mayor.

In [ ]:
# Escriba la solución aquí

## Ejercicio 023

### Escribir un programa que permita ingresar tres números enteros y mostrar el mayor el menor y el valor que esta en medio.

### Ejemplo: Si se ingresan los números 5, 3 y 7, el programa debe mostrar el número 5 como el menor, el número 7 como el mayor y el número 3 como el que esta en medio.

### Otra vez se mezclaron las instrucciones, ¿podrías arreglarlas? 

numero_uno = int(input("Ingrese el primer número: "))  
numero_dos = int(input("Ingrese el segundo número: "))  
numero_tres = int(input("Ingrese el tercer número: "))  

if medio < menor:  
    auxiliar = medio  
    medio = menor  
    menor = auxiliar  

if numero_tres > mayor:  
    mayor = numero_tres  
    medio = numero_uno  
    menor = numero_dos  

if numero_dos > mayor:  
    mayor = numero_dos  
    medio = numero_uno  
    menor = numero_tres  

mayor = numero_uno  
medio = numero_dos  
menor = numero_tres  

print("El mayor es: ", mayor)  
print("El medio es: ", medio)  
print("El menor es: ", menor)  

In [ ]:
# Escriba la solución aquí



## Ejercicio 024

### Para acceder a cierta atracción, es necesario cumplir con dos requisitos: tener __al menos__ 10 años de edad y medir __más__ de 1,60 metros.

<center>(Ojo, tener en cuenta las palabras: más,al menos y sobre todo la letra y)</center>

### Escribir un programa en Python que solicite al usuario su edad y estatura, y que determine si cumple con los requisitos para subir a la atracción. Si cumple con ambos requisitos, el programa debe imprimir "¡Puede acceder!" en la pantalla. Si no cumple con alguno de los requisitos, el programa debe imprimir un mensaje que indique el motivo por el cual no puede subir. Por ejemplo, si no cumple con el requisito de la edad, el programa debe imprimir "Lo siento, eres demasiado joven para acceder." Si no cumple con el requisito de la estatura, el programa debe imprimir "Lo siento, eres demasiado bajo para acceder"


Los conectores lógicos "and" y "or" son operadores booleanos utilizados en programación y en lógica matemática para combinar o unir dos o más expresiones lógicas.

El conector "and" se utiliza para combinar dos o más expresiones lógicas y evaluarlas en conjunto. La expresión completa es verdadera sólo si todas las expresiones individuales son verdaderas. Por ejemplo, la expresión lógica "A and B" será verdadera sólo si tanto A como B son verdaderos. Si alguna de las expresiones es falsa, la expresión completa será falsa.

La tabla de verdad del operador "and" se utiliza en lógica proposicional para determinar el valor de verdad de una proposición compuesta formada por dos proposiciones simples unidas por "and". La tabla de verdad del "and" se construye evaluando todas las posibles combinaciones de valores de verdad de las proposiciones simples, y determinando el valor de verdad de la proposición compuesta para cada combinación.

La tabla de verdad del "and" es la siguiente:

<center>

| P | Q | P and Q |
|:-:|:-:|:-------:|
| V | V |    V    |
| V | F |    F    |
| F | V |    F    |
| F | F |    F    |

</center>

En resumen, el conector "and" se utiliza para crear expresiones lógicas que requieren que todas las condiciones sean verdaderas para que la expresión sea verdadera.

In [ ]:
# Escriba la solución aquí

## Ejercicio 025

### Para acceder a cierta atracción, alcanza con que se cumpla solamente una de las siguientes condiciones: ser __mayor__ de 6 años o medir __más__ de 1,50 metros.
### Escribir un programa en Python que solicite al usuario su edad y estatura, y que determine si cumple con los requisitos para subir a la atracción. Si cumple con al menos una de las condiciones, el programa debe imprimir "¡Puede acceder!" en la pantalla. Si no cumple con ninguna de las condiciones, el programa debe imprimir un mensaje que lo indique.

<center>(Ojo, tener en cuenta las palabras: más, mayor y sobre todo la letra o)</center>

 El conector "or" se utiliza para combinar dos o más expresiones lógicas y evaluarlas en conjunto. La expresión completa es verdadera si al menos una de las expresiones individuales es verdadera. Por ejemplo, la expresión lógica "A or B" será verdadera si al menos una de las variables A o B es verdadera. Solo si ambas variables son falsas, entonces la expresión completa será falsa.

La tabla de verdad del "or" es la siguiente:

<center>

| P | Q | P or Q |
|:-:|:-:|:------:|
| V | V |    V   |
| V | F |    V   |
| F | V |    V   |
| F | F |    F   |

</center>

El conector "or" se utiliza para crear expresiones lógicas que requieren que al menos una condición sea verdadera para ser verdadera.

In [ ]:
# Escriba la solución aquí


## Ejercicio 026

### Escribir un programa que permita ingresar la cantidad de invitados a una fiesta y la cantidad de asientos disponibles en el salon. Debes indicar si alcanzan los asientos, Si los asientos no alcanzaran indicar cuántos faltan para que todos los invitados puedan sentarse.

In [ ]:
# Escriba la solución aquí


## Ejercicio 027

### Escribir un programa que permita ingresar una edad (entre 1 y 120 años), un género ('F'para mujeres, 'M' para hombres) y un nombre. En caso de haber ingresado valores erróneos (edad fuera de rango o género inválido), informar tal situación indicando el nombre de la persona. Si los datos están bien ingresados el programa debe indicar, sabiendo que las mujeres se jubilan con 60 años o más y los hombres con 65 años o más, si la persona está en edad de jubilarse.



In [ ]:
# Escriba la solución aquí


## Ejercicio 028

### Crear un programa que pida un número de mes (ejemplo 4) y escriba el nombre del mes en letras ("abril"). Verificar que el mes sea válido e informar en caso que no lo sea.

In [ ]:
# Escriba la solución aquí


## Ejercicio 029

### Escribir un programa que permita Ingresar las notas de los dos parciales de un alumno e indicar si promocionó, aprobó o debe recuperar. Si el valor de la nota no esta entre 0 y 10, debe informar un error.
- Se promociona cuando las notas de ambos parciales son mayores o iguales a 7.
- Se aprueba cuando las notas de ambos parciales son mayores o iguales a 4.
- Se debe recuperar cuando al menos una de las dos notas es menor a 4.


In [6]:
# Escriba la solución aquí

nota1 = int(input("ingrese la nota del primer parcial"))
nota2 = int(input("ingrese la nota del segundo parcial"))

notas_ok =  0 <= nota1 <= 10 and 0 <= nota2 <= 10
notas_ok =  nota1 >= 0 and nota1 <= 10 and nota2 >= 0 and nota2 <= 10
if notas_ok:
    if nota1 >= 7 and nota2 >= 7:
        print("Promociona")
    elif nota1 >= 4 and nota2 >= 4:
        print("Aprueba")
    else:
        print("Recupera")
else:
    print("error en notas")



error en notas


## Ejercicio 030

### Escribir un programa que permita al usuario ingresar dos números enteros. La computadora debe indicar si el mayor es divisible por el menor. 

(Un número entero a es divisible por un número entero b cuando el resto de la división entre a y b es 0) 

In [ ]:
# Escriba la solución aquí


## Ejercicio 031

### Una editorial determina el precio de un libro según la cantidad de páginas que contiene. El costo básico del libro es de $1000, más $35,10 por página con encuadernación rústica. Si el número de páginas supera las 300 la encuadernación debe ser especial, lo que incrementa el costo en $1200. Además, si el número de páginas sobrepasa las 600 se hace necesario otro procedimiento de encuadernación que incrementa el costo en $880. Desarrollar un programa que calcule el costo de un libro dado el número de páginas.

En Python no existen constantes como tal, pero se utiliza una convención de nomenclatura en mayúsculas para indicar que una variable no debe ser modificada. Esto se conoce como "constante simbólica" o "constante convencional". Para definir una constante convencional, simplemente se define una variable con un nombre en mayúsculas.

Por ejemplo, para el problema anterior, se pueden definir las constantes:  

COSTO_BASICO con valor 1000  

COSTO_POR_PAGINA con valor 35.10  

COSTO_ENC_RUSTICA con valor 1200  

COSTO_ENC_ESPECIAL con valor 880.  


Es una buena práctica utilizar constantes para almacenar valores fijos en un programa, ya que facilita la lectura del código y su mantenimiento, evita errores por la modificación accidental de valores y permite un fácil ajuste de los valores fijos en caso de ser necesario.


In [ ]:

# constantes simbólicas
COSTO_BASICO = 1000
COSTO_POR_PAGINA = 35.10
COSTO_ENC_RUSTICA = 1200
COSTO_ENC_ESPECIAL = 880

num_paginas = int(input("Ingrese el número de páginas del libro: "))
costo = COSTO_BASICO + (COSTO_POR_PAGINA * num_paginas)

# Escriba lo que falta de la solución aquí

print("El costo del libro es: $", costo)

## Ejercicio 032

### Una remisería requiere un sistema que calcule el precio de un viaje a partir de la cantidad de km que desea recorrer el usuario.
### Tiene la siguiente tarifa: 

- Viaje mínimo $50  

- Si recorre entre 0 y 10km: $20/km  

- Si recorre 10km o más: $15/km  


### Escribir un programa que permita ingresar la cantidad de km que desea recorrer el usuario y muestre el precio del viaje.

In [ ]:
# Escriba la solución aquí


## Ejercicio 033

### La farmacia Sindical efectúa descuentos a sus afiliados según el importe de la compra con la siguiente escala: 

- ### Menor de $5500.0 el descuento es del 4.5%  
- ### Entre $5500.0 y $10000.0 el descuento es del 8%  
- ### Más de $10000.0 el descuento es del 10.5%  

### Escribir un programa que reciba un importe e informe: el descuento y el precio neto a cobrar, con mensajes aclaratorios.

In [ ]:
# Escriba la solución aquí

## Ejercicio 034

### Escribir un programa que calcule y muestre el sueldo neto de un empleado en base a su sueldo básico y su antigüedad en años. Si es soltero se le incrementa el sueldo en 5% del salario bruto por cada año de antigüedad, mientras que si es casado se le incrementa el sueldo en 7% del salario bruto por cada año de antigüedad. También se le realizan los siguientes descuentos:

- Jubilación: 11%  

- Obra Social: 3%  

- Sindicato: 3%  

### Como datos de entrada se ingresa por teclado el sueldo básico, antigüedad y estado civil (S: Soltero / C: Casado). Se debe informar: (reemplazando los 9 por los valores que correspondan)

Estado Civil: Soltero/Casado
Sueldo básico: $ 999.99
Antigüedad: 99 años

Descuentos:

- Jubilación - 999,99  

- Obra Social - 999,99  

- Sindicato - 999,99  

Sueldo Neto 999,99




In [ ]:
# Escriba la solución aquí

## Ejercicio 035

### Existen dos reglas que identifican dos conjuntos de valores:  
- ### a) El número es de un solo dígito.  
- ### b) El número es impar.  

### A partir de estas reglas, escribir un programa que permita ingresar un número entero.
### Debe asignar el valor que corresponda a las variables booleanas:  

- esDeUnSoloDigito  
- esImpar  
- estaEnAmbos  
- noEstaEnNinguno  

### el valor Verdadero o Falso, según corresponda, para indicar si el valor número ingresado pertenece o no a cada conjunto. Definir un lote de prueba de varios números y probr el algoritmo, escribiendo los resultados.



In [13]:
# Escriba la solución aquí



esDeUnSoloDigito: True
esImpar: True
estaEnAmbos: True
noEstaEnNinguno: False


## Ejercicio 036

### Escribir un programa que permita ingresar dos números enteros y la operación a realizar('+', '-', '*', '/'). Debe mostrarse el resultado para la operación ingresada. Considerar que no se puede dividir por cero (en ese caso mostrar el texto 'ERROR').

In [ ]:
# Escriba la solución aquí


***
# <center>Flujo de repetición.</center>
***

## Ejercicio 037

### Escribir un programa que muestre todos los números enteros del 1 al 5, y luego los mismos números pero en orden inverso.

In [ ]:
# Escriba la solución aquí


## Ejercicio 038

### Escribir un programa que permita ingresar un número entero n. Debe mostrar los primeros 10 múltiplos de n.
### Ejemplo  
n = 5  

n x 1 = 5  
n x 2 = 10  
n x 3 = 15  
n x 4 = 20  
n x 5 = 25  
n x 6 = 30  
n x 7 = 35  
n x 8 = 40  
n x 9 = 45  
n x 10 = 50   



In [ ]:
# Escriba la solución aquí


## Ejercicio 039

### Escribir un programa para calcular e imprimir la suma de los números comprendidos entre 42 y 176

In [ ]:
# Escriba la solución aquí


## Ejercicio 040

### Escribir un programa que permita ingresar dos numeros enteros a y b. El programa debe mostrar:

- La suma de los numeros pares entre a y b.  
- El producto de los numeros impares entre a y b.

In [ ]:
# Escriba la solución aquí


## Ejercicio 041

### Escribir un programa que lea números enteros hasta que se ingrese un 0, y muestre el máximo.

In [ ]:
# Escriba la solución aquí

## Ejercicio 042

### Escribir un programa que lea números enteros hasta que se ingrese un 0, y muestre el promedio de los números ingresados.

In [ ]:
# Escriba la solución aquí


## Ejercicio 043

### Escribir un programa que lea números enteros hasta que se la suma de éstos sea mayor que 100, y muestre la cantidad de números ingresados.

In [ ]:
# Escriba la solución aquí


## Ejercicio 044

### Escribir un programa que permita leer dos números A y B (enteros positivos). Calcular el producto A * B por sumas sucesivas e imprimir el resultado. 

### Ejemplo:

- 4*3 = 4 + 4 + 4 (4 sumado 3 veces).  
- 3*4 = 3 + 3 + 3 + 3 (3 sumado 4 veces).

In [ ]:
# Escriba la solución aquí


## Ejercicio 045

### Escribir un programa que permita leer dos números naturales A y B. Calcular A elevado a la B mediante productos sucesivos y mostrar el resultado.  

### Ejemplo:  

- 4^3 = 4 * 4 * 4 (4 multiplicado 3 veces).  
- 3^4 = 3 * 3 * 3 * 3 (3 multiplicado 4 veces).

In [ ]:
# Escriba la solución aquí


## Ejercicio 046

### Escribir un programa que a partir de un número entero cant ingresado por el usuario permita cargar por teclado cant números enteros. La computadora debe mostrar cuál fue el mayor número y en qué posición apareció.

In [ ]:
# Escriba la solución aquí


## Ejercicio 047

### Escribir un programa que permita validar la nota de un examen. Se espera que la nota que el usuario ingrese sea un número comprendido entre 0 y 10.  
### La misma debe ser ingresada tantas veces como sea necesario hasta que quede comprendida dentro del rango indicado.

In [ ]:
# Escriba la solución aquí

## Ejercicio 048

### Escribir un programa que permita realizar varias operaciones matemáticas ingresando un caracter por cada una la operación a realizar (‘+’, ‘-’, ‘*’, ‘/’, ‘F’) y dos números enteros en el caso que no haya elegido ‘F’. La computadora debe mostrar el resultado para la operación ingresada. Considerar que no se puede dividir por cero. Cuando la operación ingresada sea ‘F’ nos indicará que no se desean calcular más operaciones.

In [ ]:
# Escriba la solución aquí


## Ejercicio 049

### Escribir un programa que permita validar una opción ingresada. Se le preguntará al usuario si desea continuar con alguna operación de la forma "¿Deseás continuar? [S/N]". Se espera que el usuario ingrese una 'S' o una 'N' (incluir las minúsculas). La opción debe ser ingresada tanto como sea necesario hasta que quede comprendida dentro de las posibilidades esperadas.

In [ ]:
# Escriba la solución aquí


## Ejercicio 050

### Escribir un programa que permita validar la nota de un examen. Se espera que la nota que el usuario ingrese sea un número comprendido entre 0 y 10.  
### La misma debe ser ingresada tantas veces como sea necesario hasta que quede comprendida dentro del rango indicado.
- Las notas 1 y 3 no usan nunca.  
- La nota 0 se reserva para los ausentes.  
### Las notas válidas pueden ser un 2 o un valor entre 4 y 10.


In [ ]:
# Escriba la solución aquí


## Ejercicio 051

### Escribir un programa que permita al usuario ingresar números hasta que se introduzca un 0.
### La computadora debe mostrar el número máximo y el número mínimo.

In [ ]:
# Escriba la solución aquí


## Ejercicio 052

### Escribir un programa que permita un programa que permita ingresar la estatura (en metros con decimales) de cada jugador de un equipo de baloncesto. La carga finalizará al ingresar cero. Calcular y mostrar la estatura promedio del equipo.

In [ ]:
# Escriba la solución aquí


## Ejercicio 053

### Escribir un programa que permita ingresar nombre y edad de un grupo de personas (para cada una, nombre y edad). La carga termina cuando en el nombre de la persona se ingresa un asterisco ('*'). Mostrar al final cómo se llama la persona más joven.

In [ ]:
# Escriba la solución aquí


## Ejercicio 054

### Escribir un programa que permita controlar con validación el ingreso a un sitio web. Teniendo dos constantes con un nombre de usuario ("admin") y una contraseña ("123456"), el programa debe permitir al usuario ingresar sus credenciales. Si las mismas son erróneas, se volverán a pedir hasta un máximo de 3 intentos. Finalmente, la computadora debe mostrar alguno de los siguientes mensajes según sea el caso: "Acceso concedido" o "Se ha bloqueado la cuenta"

In [ ]:
# Escriba la solución aquí
USUARIO = "admin"
CONTRASENA = "1234"


## Ejercicio 055

### Escribir un programa que permita para cada cliente que va al banco "Express".
### Cada cliente indica su número de documento y aguarda a ser atendido, cuando finaliza la atención del día se ingresa -1 para indicar que no hay más clientes para ser atendidos. El banco desea saber quién fue el primer cliente atendido y quién fue el último.

In [ ]:
# Escriba la solución aquí


## Ejercicio 056

### Escribir un programa que permita Leer números que representan edades de un grupo de personas, finalizando la lectura cuando se ingrese el número 999. Imprimir cuántos son menores de 18 años, cuántos tienen 18 años o más y el promedio de edad de ambos grupos. Descartar valores que no representan una edad válida. (Se considera válido una edad entre 0 y 100)

In [ ]:
# Escriba la solución aquí


## Ejercicio 057

### Escribir un programa que permita ingresar los números de legajo de los alumnos de un curso y su nota de examen final. El fin de la carga se determina ingresando un -1 en el legajo. 
### Informar para cada alumno si aprobó o no el examen considerando que se aprueba con nota mayor o igual a 4. Se debe validar que la nota ingresada sea entre 1 y 10. Terminada la carga de datos, informar:
- Cantidad de alumnos que aprobaron (nota >= 4).
- Cantidad de alumnos que desaprobaron el examen (nota < 4).
- Porcentaje de alumnos que están aplazados (nota == 1).

In [ ]:
# Escriba la solución aquí


## Ejercicio 058

### Una empresa aplica el siguiente procedimiento en la comercialización de sus productos:
- ### Aplica el precio base a la primera docena de unidades.  
- ### Aplica un 10% de descuento a todas las unidades entre 13 y 100.  
- ### Aplica un 25% de descuento a todas las unidades por encima de las 100.
### Por ejemplo, supongamos que vende 230 unidades de un producto cuyo precio base es 100. El cálculo resultante sería:  
100 * 12 + 90 * 88 + 75 * 130 = 18870, y el precio promedio será 18870 / 230 = 82,04  
### Escribir un programa que lea la cantidad solicitada de un producto y su precio base, y muestre el valor total de la venta y el precio promedio por unidad. El fin de carga se determina ingresando -1 como cantidad solicitada.
### Al finalizar informar:  
- a) Cantidad de ventas realizadas total.
- b) Cantidad de ventas que se aplicaron un 10% de descuento.
- c) Cantidad de ventas que SOLO se aplicó el precio base, no se le realizaron descuentos

In [ ]:
# Escriba la solución aquí


## Ejercicio 059

### Escribir un programa para un negocio de venta de granos que desea controlar las ventas realizadas. De cada venta ingresa el importe total y un código que indica la forma de pago. Los códigos puede ser:
- C: cheque, 20% de recargo.  
- E: efectivo, 10% de descuento.  
- T: con tarjeta, 12% de recargo.  
### Se debe ingresar una F para finalizar el día de venta y arrojar los siguientes totales.
| Forma de Pago    | Total     |  
| ---------------- | --------- |
| Efectivo en Caja | $ xxxx.xx |  
| Tarjeta/Crédito  | $ xxxx.xx |  
| Cheques          | $ xxxx.xx |  
| Total de Venta   | $ xxxx.xx |  
| Importe del IVA  | $ xxxx.xx |  



In [ ]:
# Escriba la solución aquí


## Ejercicio 060

### Escribir un programa para calcular el sueldo final a pagar a cada empleado de una empresa. De cada uno se tiene, sueldo básico, antigüedad, cantidad de hijos y estudios superiores (‘S’ o ‘N’). Además, se conocen los porcentajes de aumento del sueldo que dependen de los siguientes factores:

- Si el empleado tiene más de 10 años de antigüedad: aumento del 10%  
- Si el empleado tiene más de 2 hijos: aumento del 10%, si solo tiene uno 5%  
- Si el empleado posee estudios superiores: aumento del 5%  

### Luego de ingresar los datos de un empleado se debe preguntar si se desea ingresar otro empleado o no. Se termina la carga cuando no se deseen ingresar más empleados.
- Determinar:  
    - a. Por cada empleado: número de empleado, el sueldo básico y el nuevo sueldo.  
    - b. Sueldo nuevo promedio de la empresa.  
    

In [ ]:
# Escriba la solución aquí


## Ejercicio 061

### Escribir un programa que permita ingresar un número entero positivo y mostrar su factorial. El factorial de un número es el producto de todos los números enteros desde 1 hasta el número ingresado. Por ejemplo, el factorial de 5 es 1 * 2 * 3 * 4 * 5 = 120.

In [ ]:
# Escriba la solución aquí


## Ejercicio 062

### Un entrenador le ha propuesto a un atleta recorrer una ruta de cinco kilómetros durante 10 días para determinar si es apto para la prueba de 5 kilómetros que se desarrollará en el Parque. No se sabe si está apto y para eso nos piden que hagamos el siguiente programa.
### Nos ingresan por cada día de entrenamiento tiempo de la prueba en minutos (entero mayor que 0 y menor a 100)
### Para considerarlo apto debe cumplir las siguientes condiciones:
- Que en ninguna de las pruebas haga un tiempo mayor a 20 minutos. 
- Que al menos en una de las pruebas realice un tiempo menor de 15 minutos.
- Que su promedio sea menor o igual a 18 minutos.
### Se pide:
- Diseñar un algoritmo para registrar los datos y decidir si es apto para la competencia.
- Sólo en caso de estar apto, informar el promedio y el número de día en el que realizó el
menor tiempo 

In [ ]:
# Escriba la solución aquí

CANTIDAD_DIAS_ENTRENAMIENTO = 10

for dia in range(CANTIDAD_DIAS_ENTRENAMIENTO):# [0,1,2,3,4,5,6,7,8,9]
    tiempo = int(input(f"Día: {dia+1} Ingrese Tiempo [0..100]: "))
    while tiempo < 0 or tiempo > 100: # MIENTRAS ERROR
        tiempo = int(input(f"Día: {dia+1} Ingrese Tiempo [0..100]: "))


## Ejercicio 063

### Escribir un programa para registrar y obtener información sobre los viajes diarios de un conductor de Uber.
### Cada vez que comienza un viaje se debe ingresar la distancia recorrida, indicando si el viaje fue corto (‘C’), mediano (‘M’), largo (‘L’) o si es el fin de los datos (‘Z’).  
### El día finaliza cuando se llega a los 30 viajes o cuando se ingresa distancia ‘Z’ (fin de los datos).

### Por cada viaje se debe ingresar la siguiente información:

- ### Cantidad de pasajeros (mayor a 0 y menor a 4).
- ### Importe a cobrar, incluyendo la el costo básico ($180).  
- ### Por cada pasajero de ese viaje se debe ingresar:  
    - ### Nombre.
    - ### Edad (mayor a 18 y menor a 120 años).  

### Se desea saber, para cada viaje, cuál es la persona más grande y su nombre.

### Al finalizar el día de trabajo, el programa debe informar:

- ### Cantidad de viajes por cada categoría (‘C’, ‘M’, ‘L’).
- ### Recaudación total.
- ### Valor promedio del viaje.
- ### Porcentaje de viajes cortos.

### Todos los datos ingresados deben ser validados.








In [ ]:
# Escriba la solución aquí


## Ejercicio 064

### Un animal en rehabilitación después de una cirugía requiere ser alimentado y monitoreado en un zoológico. Se alimenta al animal 3 veces al día y se le da de comer hasta que ya no tenga ganas de comer.
### Por cada tanda de comida, se debe ingresar la cantidad de alimento en kg (número entero) que se le dio y se le debe preguntar al animal si quiere seguir comiendo ('S', 'N').
### Se desea conocer:
- ### Cuál de las 3 comidas fue la que el animal comió más cantidad de alimento y cuánto fue esa cantidad.
- ### El total en kg de alimento recibido.
- ### Promedio de alimento por tanda.

In [ ]:
# Escriba la solución aquí


## Ejercicio 065

### Una cadena de comida rápida solicita el desarrollo de una aplicación para sus terminales de autoservicio que permita a los clientes armar su propio menú. Los clientes pueden elegir entre diferentes opciones de combos o pedir por separado la comida, bebida y postre.

### Al iniciar la aplicación, se debe mostrar el siguiente menú de opciones:

- #### 1) Combo 1: Hamburguesa, papas fritas y gaseosa - $1550
- #### 2) Combo 2: Hamburguesa con queso, papas fritas y gaseosa - $1650
- #### 3) Hamburguesa sola - $1300
- #### 4) Hamburguesa con queso - $1400
- #### 5) Gaseosa - $700
- #### 6) Postre - $600
- #### 7) Agregar aderezo - $100
- #### 8) Terminar

### Luego de seleccionar cada ítem, se debe mostrar el subtotal para que el cliente pueda decidir si desea agregar más productos a su pedido antes de terminar.  
### El valor mínimo del pedido debe ser de $1550. Si el cliente elige un combo, se debe sumar el importe total al subtotal. Si el cliente selecciona opciones 3 a 7, se le debe preguntar la cantidad deseada y calcular el valor total antes de sumarlo al subtotal.
### Al finalizar el pedido, se debe mostrar el monto total a pagar, el ítem más caro y, si no se han seleccionado productos, mostrar un mensaje que diga "Pedido cancelado".

In [ ]:
# Escriba su solución aquí


## Ejercicio 066a

### La biblioteca de la ciudad necesita organizar y hacer un recuento de los libros que tiene en sus 5 estantes. Para cada uno de los 5 estantes, se deben ingresar los libros, y para cada libro, se debe ingresar su nombre (usando 'FIN' si no hay más libros para ese estante), género (usando 'I' para Infantil, 'N' para Novela, o 'H' para Historia), y cantidad de páginas (mayor a 0). Una vez que se han ingresado los datos de 1 estante, se debe mostrar por pantalla el nombre del libro con la mayor cantidad de páginas y su cantidad correspondiente. Al finalizar el ingreso de datos de todos los estantes, se deben mostrar la cantidad de libros por género y el promedio de libros por estante.

In [ ]:
# Escriba su solución aquí


## Ejercicio 066b

### La biblioteca de la ciudad necesita organizar y hacer un recuento de los libros que tiene en sus estantes. Para cada uno de los estantes (usando 'F' para indicar el fin de los estantes), se deben ingresar 15 libros, y para cada libro, se debe ingresar su nombre, género (usando 'I' para Infantil, 'N' para Novela, o 'H' para Historia), y cantidad de páginas (mayor a 0). Una vez que se han ingresado los datos de 1 estante, se debe mostrar por pantalla el nombre del libro con la mayor cantidad de páginas y su cantidad correspondiente. Al finalizar el ingreso de datos de todos los estantes, se deben mostrar la cantidad de libros por género.

In [ ]:
# Escriba su solución aquí


## Ejercicio 066c

### La biblioteca de la ciudad necesita organizar y hacer un recuento de los libros que tiene en sus estantes. Para cada uno de los estantes (usando 'F' para indicar el fin del estante), se deben ingresar los libros, y para cada libro, se debe ingresar su nombre (usando 'FIN' si no hay más libros para ese estante), género (usando 'I' para Infantil, 'N' para Novela, o 'H' para Historia), y cantidad de páginas (mayor a 0). Una vez que se han ingresado los datos de 1 estante, se debe mostrar por pantalla el nombre del libro con la mayor cantidad de páginas y su cantidad correspondiente. Al finalizar el ingreso de datos de todos los estantes, se deben mostrar la cantidad de libros por género y el promedio de libros por estante.

In [ ]:
# Escriba su solución aquí


## Ejercicio 066d

### La biblioteca de la ciudad necesita organizar y hacer un recuento de los libros que tiene en sus estantes. Para cada uno de los 5 estantes, se le pide al usuario que ingrese la cantidad de libros que tendrá ese estante. Para cada libro, se debe ingresar su nombre, género (usando 'I' para Infantil, 'N' para Novela, o 'H' para Historia), y cantidad de páginas (mayor a 0). Una vez que se han ingresado los datos de 1 estante, se debe mostrar por pantalla el nombre del libro con la mayor cantidad de páginas y su cantidad correspondiente. Al finalizar el ingreso de datos de todos los estantes, se deben mostrar la cantidad de libros por género.

In [ ]:
# Escriba su solución aquí
